### PR TITLE
[WIP] Add wicked_risky testing module

### PR DESCRIPTION
**Pull request by asmorodskyi**

### CODE REVIEW

File: lib/main_common.pm [hunk 1]
```diff
@@ -612,7 +612,12 @@ sub load_filesystem_tests {
 
 sub load_wicked_tests {
     loadtest "console/wicked_before_test";
-    loadtest "console/wicked_basic";
+    if (check_var('WICKED', '1')) {
```
Comment (okurz) [inline lib/main_common.pm]: Does this really make sense? Why should we call "wicked_risky" if !WICKED?

File: tests/console/wicked_before_test.pm [hunk 2]
```diff
@@ -13,17 +13,27 @@
...
     assert_script_run('sed -e "s/^WICKED_DEBUG=.*/WICKED_DEBUG=\"all\"/g" -i /etc/sysconfig/network/config');
     assert_script_run('sed -e "s/^WICKED_LOG_LEVEL=.*/WICKED_LOG_LEVEL=\"debug\"/g" -i /etc/sysconfig/network/config');
     assert_script_run('cat /etc/sysconfig/network/config');
+    my $ip;
+    if (check_var('WICKED', 'RISKY_REF')) {
+        $ip = '10.0.2.10/15';
+    }
+    elsif (check_var('WICKED', 'RISKY_SUT')) {
+        $ip = '10.0.2.11/15';
+    }
```
Comment (okurz) [inline tests/console/wicked_before_test.pm]: I suggest either

```perl
my %ip = (
    RISKY_REF => '10.0.2.10/15',
    RISKY_SUT => '10.0.2.11/15,
);
my $ip = $ip{get_required_var('WICKED')};
```

or at least an explicit `else`

Comment (okurz) [inline tests/console/wicked_before_test.pm]: I suggest either

```perl
my %ip = (
    RISKY_REF => '10.0.2.10/15',
    RISKY_SUT => '10.0.2.11/15,
);
my $ip = $ip{get_required_var('WICKED')};
```

or at least an explicit `else`

File: tests/console/wicked_risky.pm [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+use utils qw(systemctl snapper_revert_system);
+
+
+sub run {
+    my ($self)   = @_;
+    my $gre_ipv4 = '';
+    my $gre_ipv6 = '';
+    type_string("#***Test 1: Create a gre interface from legacy ifcfg files***\n");
+    if (check_var('WICKED', 'RISKY_REF')) {
+        $gre_ipv4 = '10.0.2.10';
```
Comment (okurz) [inline tests/console/wicked_risky.pm]: you are repeating the ip addresses. Better define them in a perl package which you use in multiple test modules or define them in a base class.

File: tests/hpc/slurm_slave.pm [hunk 4]
```diff
@@ -24,7 +24,7 @@ sub run {
     my $master_ip                 = get_required_var('HPC_MASTER_IP');
 
     select_console 'root-console';
-    $self->setup_static_mm_network();
+    $self->setup_static_network(get_required_var('HPC_HOST_IP'));
```
Comment (okurz) [inline tests/hpc/slurm_slave.pm]: Better extract a helper function in an HPC base class

File: lib/main_common.pm (hunk 1 above)
Comment (asmorodskyi) [reply lib/main_common.pm]: probably we can rename it to "basic" ? does it make more sense then ?

File: tests/console/wicked_risky.pm (hunk 3 above)
Comment (asmorodskyi) [reply tests/console/wicked_risky.pm]: my concern here is  data/wicked/ifcfg-gre1_ref and data/wicked/ifcfg-gre1_sut. By moving this IP's  out from wicked_risky and wicked_before_test I will not solve the problem of "having same data duplicated in several places". In my opinion it not worth creation of another perl file which only purpose would be keep 2 IPs.

File: lib/main_common.pm (hunk 1 above)
Comment (okurz) [reply lib/main_common.pm]: Like check_var ('WICKED', 'basic') maybe

Comment (okurz) [reply lib/main_common.pm]: Then you can actually call loadtest 'wicked_' . get_required_var('WICKED');

File: lib/utils.pm [hunk 5]
```diff
@@ -1388,6 +1390,21 @@ sub get_x11_console_tty {
...
+    my ($self, $ip) = @_;
+    configure_default_gateway();
+    configure_static_ip($ip);
+    configure_static_dns(get_host_resolv_conf());
+
+    # check if gateway is reachable
+    assert_script_run "ping -c 1 10.0.2.2 || journalctl -b --no-pager > /dev/$serialdev";
+}
+
+
```
Comment (okurz) [inline lib/utils.pm]: That lib/utils.pm is abused too much. Please put it into a class

File: tests/console/wicked_risky.pm (hunk 3 above)
Comment (okurz) [reply tests/console/wicked_risky.pm]: Then grep it from these data files?

**Comment by asmorodskyi** [comment]
Unfortunetly I mess up this branch somehow. To not start long fight with git I just cherry-pick changes and created a new PR. So see you in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3813

**Pull request closed without merging**